File: app/src/main/java/com/aar/app/wordsearch/di/modules/DataSourceModule.java
package com.aar.app.wordsearch.di.modules;

import android.content.Context;

import com.aar.app.wordsearch.data.sqlite.DbHelper;
import com.aar.app.wordsearch.data.sqlite.GameDataSQLiteDataSource;
import com.aar.app.wordsearch.data.xml.WordXmlDataSource;
import com.aar.app.wordsearch.data.GameDataSource;
import com.aar.app.wordsearch.data.WordDataSource;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by abdularis on 18/07/17.
 */

@Module
public class DataSourceModule {

    @Provides
    @Singleton
    DbHelper provideDbHelper(Context context) {
        return new DbHelper(context);
    }

    @Provides
    @Singleton
    GameDataSource provideGameRoundDataSource(DbHelper dbHelper) {
        return new GameDataSQLiteDataSource(dbHelper);
    }

//    @Provides
//    @Singleton
//    WordDataSource provideWordDataSource(DbHelper dbHelper) {
//        return new WordSQLiteDataSource(dbHelper);
//    }

    @Provides
    @Singleton
    WordDataSource provideWordDataSource(Context context) {
        return new WordXmlDataSource(context);
    }

}

File: app/src/main/java/co/africanwolf/hiddenwords/di/DataSourceModule.kt
package co.africanwolf.hiddenwords.di

import android.content.Context
import co.africanwolf.hiddenwords.data.WordDataSource
import co.africanwolf.hiddenwords.data.sqlite.DbHelper
import co.africanwolf.hiddenwords.data.sqlite.GameDataSQLiteDataSource
import co.africanwolf.hiddenwords.data.xml.WordXmlDataSource
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.qualifiers.ApplicationContext
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
object DataSourceModule {

    @Provides
    @Singleton
    fun provideGameRoundDataSource(dbHelper: DbHelper): GameDataSQLiteDataSource {
        return GameDataSQLiteDataSource(dbHelper)
    }

    //    @Provides
    //    @Singleton
    //    WordDataSource provideWordDataSource(DbHelper dbHelper) {
    //        return new WordSQLiteDataSource(dbHelper);
    //    }
    @Provides
    @Singleton
    fun provideWordDataSource(@ApplicationContext context: Context): WordDataSource {
        return WordXmlDataSource(context)
    }
}
